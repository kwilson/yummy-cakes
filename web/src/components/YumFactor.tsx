import * as React from 'react';

export interface YumFactorStateProps {
  value: number;
}

const YumFactor: React.StatelessComponent<YumFactorStateProps> = (props) => {
  return (
    <p>YUM: {props.value}</p>
  );
};

export default YumFactor;
