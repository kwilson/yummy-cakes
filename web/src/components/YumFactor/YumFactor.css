.yum-factor__rating {
    color: gold;
    font-size: 40px;
}
