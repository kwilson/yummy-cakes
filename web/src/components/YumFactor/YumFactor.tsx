import * as React from 'react';
import './YumFactor.css';

export interface YumFactorStateProps {
  value?: number;
}

const YumFactor: React.StatelessComponent<YumFactorStateProps> = ({ value }) => {
  if (!value) {
    return null;
  }

  const ratingLabel = `${value} out of 5`;
  const stars = '★★★★★☆☆☆☆☆'.substr(5 - value, 5);

  return (
    <div className="yum-factor">
      Yum Factor
      
      <div className="yum-factor__rating" title={ratingLabel} aria-label={ratingLabel}>
        {stars}
      </div>
    </div>
  );
};

export default YumFactor;
