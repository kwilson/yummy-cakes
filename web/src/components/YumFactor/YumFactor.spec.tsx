import * as React from 'react';
import { mount } from 'enzyme';
import { create as createSnapshot } from 'react-test-renderer';

import YumFactor from './YumFactor';

describe('YumFactor component', () => {
  it('renders null if there is no value', () => {
    // Arrange
    const cmp = <YumFactor />;

    // Act
    const el = mount(cmp);
    const snapshot = createSnapshot(cmp).toJSON();

    // Assert
    expect(el.html()).toBe(null);
    expect(snapshot).toMatchSnapshot();
  });

  it('renders null if the value is 0', () => {
    // Arrange
    const cmp = <YumFactor value={0} />;

    // Act
    const el = mount(cmp);
    const snapshot = createSnapshot(cmp).toJSON();

    // Assert
    expect(el.html()).toBe(null);
    expect(snapshot).toMatchSnapshot();
  });

  it('renders null if the value is 6', () => {
    // Arrange
    const cmp = <YumFactor value={6} />;

    // Act
    const el = mount(cmp);
    const snapshot = createSnapshot(cmp).toJSON();

    // Assert
    expect(el.html()).toBe(null);
    expect(snapshot).toMatchSnapshot();
  });

  it('renders the correct stars for each value', () => {
    // Arrange
    const testCases = [
      { value: 1, expected: '★☆☆☆☆' },
      { value: 2, expected: '★★☆☆☆' },
      { value: 3, expected: '★★★☆☆' },
      { value: 4, expected: '★★★★☆' },
      { value: 5, expected: '★★★★★' }
    ];

    testCases.forEach(testCase => {
      // Arrange
      const cmp = <YumFactor value={testCase.value} />;

      // Act
      const el = mount(cmp);
      const snapshot = createSnapshot(cmp).toJSON();

      // Assert
      const rating = el.find('.yum-factor__rating');
      expect(rating.text().length).toBe(5);
      expect(rating.props().title).toBe(`${testCase.value} out of 5`);

      expect(rating.text()).toBe(testCase.expected);

      expect(snapshot).toMatchSnapshot();
    });
  });
});
