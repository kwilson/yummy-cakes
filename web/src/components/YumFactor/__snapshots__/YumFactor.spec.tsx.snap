// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YumFactor component renders null if the value is 0 1`] = `null`;

exports[`YumFactor component renders null if the value is 6 1`] = `null`;

exports[`YumFactor component renders null if there is no value 1`] = `null`;

exports[`YumFactor component renders the correct stars for each value 1`] = `
<div
  className="yum-factor"
>
  Yum Factor
  <div
    aria-label="1 out of 5"
    className="yum-factor__rating"
    title="1 out of 5"
  >
    ★☆☆☆☆
  </div>
</div>
`;

exports[`YumFactor component renders the correct stars for each value 2`] = `
<div
  className="yum-factor"
>
  Yum Factor
  <div
    aria-label="2 out of 5"
    className="yum-factor__rating"
    title="2 out of 5"
  >
    ★★☆☆☆
  </div>
</div>
`;

exports[`YumFactor component renders the correct stars for each value 3`] = `
<div
  className="yum-factor"
>
  Yum Factor
  <div
    aria-label="3 out of 5"
    className="yum-factor__rating"
    title="3 out of 5"
  >
    ★★★☆☆
  </div>
</div>
`;

exports[`YumFactor component renders the correct stars for each value 4`] = `
<div
  className="yum-factor"
>
  Yum Factor
  <div
    aria-label="4 out of 5"
    className="yum-factor__rating"
    title="4 out of 5"
  >
    ★★★★☆
  </div>
</div>
`;

exports[`YumFactor component renders the correct stars for each value 5`] = `
<div
  className="yum-factor"
>
  Yum Factor
  <div
    aria-label="5 out of 5"
    className="yum-factor__rating"
    title="5 out of 5"
  >
    ★★★★★
  </div>
</div>
`;
